internal/inventory/gcpfetcher/fetcher_assets.go

```diff
@@ -125,54 +123,37 @@ func getAssetEvent(classification inventory.AssetClassification, item *gcpinvent
 	)
 }
 
-func findRelatedAssetIds(t inventory.AssetType, item *gcpinventory.ExtendedGcpAsset) []string {
+func findRelatedAssetIds(item *gcpinventory.ExtendedGcpAsset) []string {
 	ids := []string{}
 	ids = append(ids, item.Ancestors...)
 	if item.Resource != nil {
 		ids = append(ids, item.Resource.Parent)
 	}
-
-	ids = append(ids, findRelatedAssetIdsForType(t, item)...)
-
+	ids = append(ids, findRelatedAssetIdsForType(item)...)
 	ids = lo.Compact(ids)
 	ids = lo.Uniq(ids)
 	return ids
 }
 
-func findRelatedAssetIdsForType(t inventory.AssetType, item *gcpinventory.ExtendedGcpAsset) []string {
+func findRelatedAssetIdsForType(item *gcpinventory.ExtendedGcpAsset) []string {
 	ids := []string{}
-
-	var fields map[string]*structpb.Value
-	if item.Resource != nil && item.Resource.Data != nil {
-		fields = item.GetResource().GetData().GetFields()
+	var pb map[string]any
+	if hasResourceData(item) {
+		pb = item.GetResource().GetData().AsMap()
 	}
 
-	switch t {
-	case inventory.AssetClassificationGcpInstance.Type:
-		if v, ok := fields["networkInterfaces"]; ok {
-			for _, networkInterface := range v.GetListValue().GetValues() {
-				networkInterfaceFields := networkInterface.GetStructValue().GetFields()
-				ids = appendIfExists(ids, networkInterfaceFields, "network")
-				ids = appendIfExists(ids, networkInterfaceFields, "subnetwork")
-			}
-		}
-		if v, ok := fields["serviceAccounts"]; ok {
-			for _, serviceAccount := range v.GetListValue().GetValues() {
-				serviceAccountFields := serviceAccount.GetStructValue().GetFields()
-				ids = appendIfExists(ids, serviceAccountFields, "email")
-			}
-		}
-		if v, ok := fields["disks"]; ok {
-			for _, disk := range v.GetListValue().GetValues() {
-				diskFields := disk.GetStructValue().GetFields()
-				ids = appendIfExists(ids, diskFields, "source")
-			}
-		}
-		ids = appendIfExists(ids, fields, "machineType")
-		ids = appendIfExists(ids, fields, "zone")
-	case inventory.AssetClassificationGcpFirewall.Type, inventory.AssetClassificationGcpSubnet.Type:
-		ids = appendIfExists(ids, fields, "network")
-	case inventory.AssetClassificationGcpProject.Type, inventory.AssetClassificationGcpBucket.Type:
+	switch item.AssetType {
+	case gcpinventory.ComputeInstanceAssetType:
+		ids = append(ids, values([]string{"networkInterfaces", "network"}, pb)...)
+		ids = append(ids, values([]string{"networkInterfaces", "subnetwork"}, pb)...)
+		ids = append(ids, values([]string{"serviceAccounts", "email"}, pb)...)
+		ids = append(ids, values([]string{"disks", "source"}, pb)...)
+		ids = append(ids, values([]string{"machineType"}, pb)...)
+		ids = append(ids, values([]string{"zone"}, pb)...)
+
+	case gcpinventory.ComputeFirewallAssetType, gcpinventory.ComputeSubnetworkAssetType:
+		ids = append(ids, values([]string{"network"}, pb)...)
+	case gcpinventory.CrmProjectAssetType, gcpinventory.StorageBucketAssetType:
```

Comment:
i'm not sure why we're adding `binding.Role` and `binding.Members` just for projects and buckets asset types. 

unrelated to this PR tho